0035-spsp-invoices/0035-spsp-invoices.md

```diff
@@ -0,0 +1,130 @@
+  1. The SPSP Client will adjust their `sendMax` to reflect the amount they're willing to send.
+      * `sendMax` SHOULD be derived from the Invoice.
+  2. The SPSP Server will adjust their `receiveMax` to reflect the amount they're willing to receive.
+      * `receiveMax` SHOULD be derived from the Invoice.
+  3. The SPSP Client's and Server's [STREAM Modules](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Definitions) will move as much value as possible while staying inside these bounds.
+  4. If the SPSP Client reaches their `sendMax`, they end the stream and the connection. If the SPSP Server reaches their `receiveMax`, they will end the stream and the connection.
+
+The STREAM parameters - `sendMax` and `receiveMax` - are defined in [STREAM's frame encoding](../0029-stream/0029-stream.md#53-frames).
+
+
+## Specification
+
+### Query (`GET <SPSP Endpoint>`)
+
+The SPSP Client queries the Invoice Payment Pointer to get information about the SPSP Server as well as Invoice specific details:
+
+#### Request
+
+``` http
+GET /invoice123 HTTP/1.1
+Host: example.com
+Accept: application/spsp4+json
+```
+
+#### Response
+
+``` http
+HTTP/1.1 200 OK
+Content-Type: application/spsp4+json
+
+{
+  "destination_account": "example.ilpdemo.red.invoice123",
+  "shared_secret": "6jR5iNIVRvqeasJeCty6C+YB5X9FhSOUPCL/5nha5Vs=",
+  "push": {
+    "balance": "5360",
+    "invoice": {
+      "totalPaymentDue": {
+        "price": 199.99,
+        "priceCurrency": "USD"
+      },
+      "paymentDueDate": "2019-01-30T00:00Z",
+      "paymentStatus": "http://schema.org/PaymentDue",
+      "description": "Chair model 'Rustic'",
+      "provider": "The Red Furniture Store",
+      "asset": {
+        "code": "USD",
+        "scale": 2
+    }
+  }
+  
+}
+```
+
+##### Response Body
+
+The response body is a JSON object that includes basic account details necessary for setting up a STREAM connection as well as optional parameters for displaying purposes. The fields are defined in the following: 
+
+| Field | Type | Description |
+|---|---|---|
+| `destination_account` | [ILP Address](../0015-ilp-addresses/0015-ilp-addresses.md) | (see [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Response-Body)) |
+| `shared_secret` | 32 bytes, [base64 encoded](https://en.wikipedia.org/wiki/Base64) (including padding) | (see [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Response-Body)) |
+| `push`  | Object | _(OPTIONAL)_ Details of this specific push payment pointer. |
+| `push.balance`  | String Integer | _(OPTIONAL)_ Amount, denoted in `push.invoice.asset.code`, which completes the invoice payment. |
+| `push.invoice` | Object | _(OPTIONAL)_ Invoice details. It SHOULD follow the invoice schema proposed by [schema.org](https://schema.org/Invoice). Additionally, it SHOULD include the followin parameters:|
```

Comment:
This makes the spec quite confusing to implement. We also don't have any dependency on schema.org in any other specs. Better to just describe the fields you used in your example and maybe just note that they came from this schema on schema.org

Reply:
What I don't like about the schma.org spec is that the amounts are not integers with scale as we have it everywhere else. 
```
"totalPaymentDue": {
        "price": 199.99,
        "priceCurrency": "USD"
      },
```
@sharafian Do you think we should "redefine" that?

Reply:
I think I'd just scrap the `totalPaymentDue`, `paymentDueDate`, `paymentStatus`, and `provider` fields. We could replace it with `amount` (which is an integer amount in units defined by `asset`, and we could add an `additional_details` field that you can put arbitrary extra information into.

Reply:
Basically, you are suggesting to not use schema.org at all but to try to come up with the fields that are absolutely necessary (maybe inspired by schema.org) and have an `additional_fields`.

Reply:
yeah, I think the fields that are absolutely necessary are just the `amount` and `asset`. Everything else can go into a `additional_fields` object because they're just metadata that describes the payment but doesn't really affect the logic of how you pay.